proposals/0307-GetInputScreen.md

```diff
@@ -102,40 +128,175 @@ This enables text input fields, numeric input fields and combo box input fields.
     <element name="NUMBER" />
       <description>
-        This is a notice for the HMI to bring up a keyboard, however the response for this field will still be a string.
+        This is a notice for the HMI to bring up a keypad
+      </description>
+  </enum>
+```
+
+and the `HMI_API`:
+```xml
+<struct name="Field">
+  <description>
+    A single field for data input
+  </description>
+  <param name="type" type="Common.FieldType" mandatory="true">
+  </param>
+  <param name="label" maxlength="500" type="String" mandatory="true">
+  </param>
+  <param name="image" type="Image" mandatory="false">
+    <description>
+      An image icon for the input field
+    </description>
+  </param>
+  <param name="text" maxlength="500" type="String" mandatory="false">
+    <description>
+      The text that is already present in the input field
+    </description>
+  </param>
+  <param name="format" maxlength="500" type="String" mandatory="false">
+    <description>
+      A text string used to format the input text to a certain style. Any characters except underscore will remain in the field with no way to type over them. The underscore character will represent a space that the user can type in.  E.g. A format string for phone number would be +_(___)___-____
+    </description>
+  </param>
+  <param name="additionalInfo" maxlength="500" type="String" mandatory="false">
+    <description>
+      More information that is shown by the input field
+    </description>
+  </param>
+  <param name="error" type="Boolean" mandatory="false">
+    <description>
+      If set to true, then the field has some problem with it and the HMI should highlight it in red.
+    </description>
+  </param>
+  <param name="comboBoxChoices" type="String" maxlength="500" array="true" minsize="1" maxsize="200" mandatory="false">
+    <description>
+      List of choices if the type is COMBO_BOX. Otherwise ignored.
+    </description>
+  </param>
+  <param name="inputMask" type="Common.KeyboardInputMask" mandatory="false">
+    <description>
+      Used to cover up personal information like credit card number. Defaults to "DISABLE_INPUT_KEY_MASK"
+    </description>
+  </param>
+</struct>
+```
+```xml
+  <enum name="FieldType">
+    <description>The type of input field</description>
+    <element name="COMBO_BOX" />
+    <element name="TEXT" />
+    <element name="NUMBER" />
+      <description>
+        This is a notice for the HMI to bring up a keypad.
       </description>
   </enum>
 ```
 
 Regarding the response which would be sent when the user presses a softbutton, this would use RPC encryption which is handled by policies.
 There would be one string response for each field. If a field was left empty, the associated string would be blank.
 ```xml
-<function name="GetInput" messagetype="response">
-  <param name="fields" type="String" maxlength="500" array="true" minsize="0" maxsize="100" mandatory="true">
+<function name="DisplayForm" messagetype="response">
```

Comment:
I believe this is addressing item 12 from the requested revisions. Can you please explicitly call out that these are changes to the Mobile API?